### DISCUSSION TITLE
Caddy question

### DISCUSSION THREAD

**Discussion by lloydbayley** [mcguirepr89/BirdNET-Pi#694] [Q&A] [answered]
G'day,

I was under the impression that caddy restricted its' answers to the url you provide in the settings page (Advanced).
Oddly-enough, I've discovered that caddy will answer from any DNS name I redirect to the IP Address and also it will answer to the IP address itself.

I don't want to go poking around too much in the caddy config file because I don't want to bugger the BirdNetPi and have been reading up on caddy but I can't get a definitive answer. Is anyone a caddy expert who could shed some light on this please?

**Comment by mcguirepr89** [comment, marked as answer]
Hi, Lloyd.

Caddy's address lines need to be configured with a specific address to limit its responses. Almost certainly your address line contains a catchall "`http://`" or "`:80`" which disables any checks against the request. I hope that saved you some time looking through the caddy docs.

**Reply by lloydbayley** [reply]
G'day Patrick! :)
Good to hear from you!

Yes, there's something going on...

I have garbled the "special" auth numbers here for privacy but this is what I have.

```
http:// https://berowra.birdnetpi.com {
        root * /home/birdpiadmin/BirdSongs/Extracted
        file_server browse
        handle /By_Date/* {
                file_server browse
        }
        handle /Charts/* {
                file_server browse
        }
        basicauth /views.php?view=File* {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        basicauth /Processed* {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        basicauth /scripts* {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        basicauth /stream {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        basicauth /phpsysinfo* {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        basicauth /terminal* {
                birdnet EdNcXpHdHQ0Z3JtOGxyQzNm1heS5IQzFIWE8u
        }
        reverse_proxy /stream localhost:8000
        php_fastcgi unix//run/php/php7.4-fpm.sock
        reverse_proxy /log* localhost:8080
        reverse_proxy /stats* localhost:8501
        reverse_proxy /terminal* localhost:8888
}
```
I've been looking at the installation/update files for birdnet to see what a new install creates. Mine is possibly full of leftovers from a bygone era, plus, I started with the Core repo so I should have a good look at this and possibly start it from scratch do you think?

Cheers,


Lloyd. :)

**Reply by mcguirepr89** [reply]
The "`http://`" portion of the address line allows any http request to receive a response. If you remove it and have only `berowra.birdnetpi.com` (without `https://`), that will enable the forced HTTPS redirect and will no longer complete requests that don't match that FQDN. BirdNET-Pi puts the "`http://`" portion in place because most users don't know enough about networking to ensure they match their address site block. That means that **if you make _any_ manual changes to the Caddyfile, you will need to be certain not to update setting using the "Tools" > "Settings" webGUI, as anytime you save settings there, your Caddyfile will be rewritten with the designed defaults.**

**Reply by lloydbayley** [reply]
Yeh, I thought as much. I'll give it some thought. Thank for the pointers! :)

**Reply by lloydbayley** [reply]
Ok. That works, looking at code and experimenting, I am all good with updating all settings except for the URL. If I touch that, its back (as we thought). I'll just remember that as it's not something that will be changing.
YAY for BirdNetPi!
